explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/TensorConverter.java

```diff
@@ -84,6 +84,16 @@ static Feature contentsToFeature(ModelInferRequest.InferInputTensor tensor, Stri
         }
     }
 
+    /**
+     * Convert a {@link ModelInferResponse.InferOutputTensor} to a {@link Output}.
+     * The datatype is inferred from the contents of the tensor, rather than the datatype field, since these can
+     * sometimes be mismatched by the model itself.
```

Comment:
Can you please add a test case with this scenario?

Reply:
Test added, confirmed to fail with previous implementation.